torch/csrc/jit/type.cpp

```diff
@@ -164,6 +164,12 @@ c10::optional<TypePtr> unifyTypes(const TypePtr& t1, const TypePtr& t2) {
     return static_cast<TypePtr>(DynamicType::get());;
   }
 
+  if (t1->isSubtypeOf(NoneType::get()) && !t2->isSubtypeOf(NoneType::get())) {
```

Comment:
Do you mind adding a comment here ? Maybe mention that the case with (Optional[T], None) -> Optional[T] is handled above this